### DISCUSSION TITLE
OS Specific build steps

### DISCUSSION THREAD

**Discussion by jlconlin** [community/community#25155] [Actions]
<p>I need to build my application on Linux and macOS. I need to do some setup up stuff before I run and the setup is different for each OS.  <strong>How can I specify the OS-dependent setup</strong>**?**</p>
<p>This is what I have so far. I get bogged down because I can’t apt-get in macOS.</p>
<pre><code>jobs:
  build:
    runs-on: ${{matrix.os}}
    strategy:
      matrix:
        os: [ubuntu-18.04, macOS]
    steps:
    - name: Install gfortran
      run: sudo apt-get install gfortran
    - uses: actions/checkout@v2
    - name: mkdir bin
      run: mkdir bin
    - name: cmake
      run: cmake ..
      working-directory: ./bin
    - name: make
      run: make -j
      working-directory: ./bin
    - name: ctest
      run: ctest -j
      working-directory: ./bin
</code></pre>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/jlconlin">@jlconlin</a> ,</p>
<p>On <strong>macOS</strong> , <strong>Xcode</strong> contains the tools needed to compile native code on the machine, include <strong>gfortran</strong>. So you do not need to additionally install <strong>gfortran</strong>  for <strong>macOS</strong> , and just install <strong>gfortran</strong> for <strong>Linux</strong>.</p>
<p>For example:</p>
<pre><code>jobs:
  test:
    name: Test
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, macos-latest]
    steps:
      - name: Install gfortran for Linux
        if: matrix.os == 'ubuntu-latest'
        run: sudo apt-get install gfortran

      - name: Check gfortran version
        run: gfortran --version
</code></pre>
